### ISSUE TITLE
Runtime error: Dynamic server usage: no-store fetch

### ISSUE THREAD

**Issue by jamesvclements** [vercel/storage#213]
I have a `utils-server.ts` file:

```ts
import { kv } from "@vercel/kv";
import "server-only";

export const getLastVisitorLocation = async () => kv.get("lastVisitorLocation");
```

My root layout calls this function to prepopulate some global state:
```tsx
export default async function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  /* Get last visitor location for server-side rendering */
  /* This is overwritten on hydration, but it's better than starting blank */
  const lastVisitorLocation = await getLastVisitorLocation();
  console.log(`Last visitor location SSR: ${lastVisitorLocation}`);
```

When I navigate from a static page to a dynamic page, I get this error:
![localhost3000hbo-Tuesday-June-13-2023-07 10 20PM@2x](https://github.com/vercel/storage/assets/20052710/67b7fd68-5eae-4e7a-8f6d-f000c7c8d452)

I can directly load the page, and it works. But when navigating from one page to another, it breaks.

**Comment by jamesvclements** [comment]
I was able to solve this for the moment by directly calling the API instead of using the SDK:

```ts
export const getLastVisitorLocation = async () => {
  const response = await fetch(
    `${process.env.KV_REST_API_URL}/get/lastVisitorLocation`,
    {
      headers: {
        Authorization: `Bearer ${process.env.KV_REST_API_TOKEN}`,
      },
    }
  );
  const data = await response.json();
  return data.result;
};
```

**Comment by TrustyTechSG** [comment]
Im also having this exact same issue, every soft navigation will have this error, only full page refresh works. hope it can be fixed soon.
![image](https://github.com/vercel/storage/assets/36871034/e52bc8f6-b7d9-4e71-bab8-1893c3fe60eb)

**Comment by andyszy** [comment]
I have this issue also

**Comment by vvo** [comment]
Hey there, we're investigating this issue and here's how we plan to resolve it:
- remove `cache` property from fetch requests
- recommend using https://nextjs.org/docs/app/api-reference/functions/unstable_noStore and https://nextjs.org/docs/app/api-reference/functions/unstable_cache for cases where you need to opt out of the default Next.js behavior

**Comment by vvo** [comment]
A fix is incoming and will be published as 1.0.0. Since this is a breaking change (major).
